src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -77,50 +78,107 @@ class WorkspaceNewRoomPage extends React.Component {
+        const policyID = this.state.policyID;
+        const visibility = this.state.visibility;
+        const errors = {};
+
+        if (!roomName || roomName === '#') {
+            errors.roomName = 'Please enter a room name';
+        } else if (
+            policyID
+            && this.checkAndModifyRoomName(roomName).isExistingRoomName
+        ) {
+            errors.roomName = this.props.translate('newRoomPage.roomAlreadyExists');
+        } else {
+            errors.roomName = '';
+        }
+
+        if (!policyID) {
+            errors.policyID = 'Please select a workspace';
+        } else {
+            errors.policyID = '';
+        }
+
+        if (!visibility) {
+            errors.visibility = 'Please select a visibility';
+        } else {
+            errors.visibility = '';
+        }
+
+        if (!errors.roomName && !errors.policyID && !errors.visibility) {
+            this.setState({errors: {}});
+            return true;
+        }
+
+        this.setState(prevState => ({
+            errors: {...prevState.errors, ...errors},
+        }));
+        return false;
+    }
+
+    /**
+     * Clear the errors associated to inputKey if found and store the inputKey new value in the state.
+     *
+     * @param {String} inputKey
+     * @param {String} value
+     */
+    clearErrorAndSetValue(inputKey, value) {
+        this.setState(prevState => ({
+            [inputKey]: value,
+            errors: {
+                ...prevState.errors,
+                [inputKey]: false,
+            },
+        }));
     }
 
     /**
      * Modifies the room name to follow our conventions:
      * - Max length 80 characters
      * - Cannot not include space or special characters, and we automatically apply an underscore for spaces
      * - Must be lowercase
-     * Also checks to see if this room name already exists, and displays an error message if so.
+     * provides the modified room name and a boolean indicating if the room name is already existing or not.
      * @param {String} roomName
      *
-     * @returns {String}
+     * @returns {Object} data
+     * @returns {String} data.modifiedRoomName
+     * @returns {Boolean} data.isExistingRoomName
      */
     checkAndModifyRoomName(roomName) {
         const modifiedRoomNameWithoutHash = roomName.substr(1)
             .replace(/ /g, '_')
             .replace(/[^a-zA-Z\d_]/g, '')
             .substr(0, CONST.REPORT.MAX_ROOM_NAME_LENGTH)
             .toLowerCase();
-        const finalRoomName = `#${modifiedRoomNameWithoutHash}`;
+        const modifiedRoomName = `#${modifiedRoomNameWithoutHash}`;
 
         const isExistingRoomName = _.some(
             _.values(this.props.reports),
-            report => report && report.policyID === this.state.policyID && report.reportName === finalRoomName,
+            report => report && report.policyID === this.state.policyID
+            && report.reportName === modifiedRoomName,
         );
```

Comment:
Let's break this into a separate function which we can use in `validate`. This function will only be used to modify and store the room name.